src/aot/aot.cpp

```diff
@@ -0,0 +1,273 @@
+#include "aot.h"
+
+#include <cstdint>
+#include <cstring>
+#include <exception>
+#include <fcntl.h>
+#include <fstream>
+#include <istream>
+#include <streambuf>
+#include <sys/mman.h>
+#include <sys/stat.h>
+
+#include <cereal/archives/binary.hpp>
+#include <cereal/types/string.hpp>
+#include <cereal/types/unordered_map.hpp>
+#include <cereal/types/vector.hpp>
+
+#include "log.h"
+#include "utils.h"
+
+static constexpr auto AOT_MAGIC = 0xA07;
+
+// AOT payload will have this header at the beginning. We don't worry about
+// versioning the header b/c we enforce that an AOT compiled script may only
```

Comment:
do you plan to keep this long term? Having to exactly match will be annoying once aot stabilizes